apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -39,6 +45,34 @@ export const getPageById = (id: string) => {
     .executeTakeFirstOrThrow()
 }
 
+export const updatePageById = (
+  page: Partial<Omit<Page, 'id' | 'blobId'>> & { id: string },
```

Comment:
hmm, why do we omit the `id` here, only to add `& { id: string }` back again ah?

Reply:
`Partial` means that the `id` will be set to optional. we don't want this behaviour so we do a union to let the id be required. another option considered but discarded was having an extra arg called `id: string`. discarded because most callers (atm 1) of `updatePageById` probably already have a object - easier to consume in this way.